canbench-bin/src/table.rs

```diff
@@ -66,47 +66,65 @@ pub(crate) fn filter_entries(data: &[Entry], noise_threshold: f64) -> Vec<Entry>
     filtered
 }
 
-pub(crate) fn print_table(data: &[Entry]) {
+pub(crate) fn print_table(data: &[Entry], max_displayed_rows: usize) {
     let columns = [
         "status", "name", "ins", "ins Δ%", "HI", "HI Δ%", "SMI", "SMI Δ%",
     ];
 
-    let mut rows = Vec::new();
-    for entry in data {
-        let name = entry.benchmark.full_name();
-        let row = [
-            entry.status.clone(),
-            name,
-            entry.instructions.fmt_human_current(), // Table report use short numbers
-            entry.instructions.fmt_human_percent(),
-            entry.heap_increase.fmt_human_current(),
-            entry.heap_increase.fmt_human_percent(),
-            entry.stable_memory_increase.fmt_human_current(),
-            entry.stable_memory_increase.fmt_human_percent(),
-        ];
-        rows.push(row);
-    }
-
-    // Calculate max column widths
-    let mut col_widths = columns
+    let mut rows: Vec<Vec<String>> = data
         .iter()
-        .map(|header| header.len())
-        .collect::<Vec<_>>();
+        .map(|entry| {
+            vec![
+                entry.status.clone(),
+                entry.benchmark.full_name(),
+                // Table report uses short numbers
+                entry.instructions.fmt_human_current(),
+                entry.instructions.fmt_human_percent(),
+                entry.heap_increase.fmt_human_current(),
+                entry.heap_increase.fmt_human_percent(),
+                entry.stable_memory_increase.fmt_human_current(),
+                entry.stable_memory_increase.fmt_human_percent(),
+            ]
+        })
+        .collect();
+
+    let total_rows = rows.len();
+
+    // Apply row limit and add omitted indicator if needed
+    if total_rows > max_displayed_rows && max_displayed_rows >= 3 {
```

Comment:
why `max_displayed_rows >= 3`? I'd drop this (since it's always true at the moment and if it became true - because max_displayed_rows is very low - then the effect will be that nothing is filtered)

Reply:
I thought it would only make sense to omit results when there are at least 3 results. A bit of safe guarding to prettify the results. But you're right, low numbers should not be put there in the first place.